Project builder documentation rewrite

jelliotartz opened the pull request:
Staging branch URL: https://project-builder-documentation-rewrite.pfe-preview.zooniverse.org/help

Continuation of #3189.
  - Related: https://github.com/zooniverse/Panoptes-Front-End/issues/1371

## Describe your changes
- Adds the new 'How to build a project' help text on the '/help' page.
  - Text styled with Markdown and rendered via a series of Markdown elements.
  - Text stored in `/locales/en`.
- Adds unit tests for the static page.
- Formatting, styling and content taken from [this word doc from original PR](https://github.com/zooniverse/Panoptes-Front-End/files/1457485/All.sections.docx).
- [Staging branch of original PR](https://projectbuilderdocumentation.pfe-preview.zooniverse.org/lab-how-to).
- ~~`project-builder-walkthrough`: This page preserves a link to the Kitteh Zoo demo/walkthrough. It is accessible via a link in the "Further Help" section, but has no nav link (See top of [original PR](https://github.com/zooniverse/Panoptes-Front-End/pull/3189) for strategy).~~ (JA edit: now added as `example.cjsx`, with nav link).
  - Happy to convert the kitteh zoo walkthrough to ES6, but I think this is best done on a separate PR.

Tagging @hrspiers and @bnewing in case they would like to review/add input, but not assigning/requesting a review from either of you. Any input is appreciated :) 

### Issues
- Marking section:
  - Has the name of this section in the Project Builder changed? is it now 'Choices'? I kinda got a little lost trying to follow this section, not sure if it needs to be updated to reflect changes in the Project Builder
  - the list of tool types is no longer accurate, and I think we should update the list before deploying this.
- Manual testing haalp plz: I've gone through and checked every link on the page, but would appreciate additional manual testing help in case I missed anything.
- Editing for content: I think it looks great, and is very clear and easy to follow; however, I'm relatively new, and perhaps a second set of eyes on the content would be helpful/beneficial before this gets deployed.

### Required Manual Testing
- [x] Does the non-logged in home page render correctly?
- [x] Does the logged in home page render correctly?
- [x] Does the projects page render correctly?
- [x] Can you load project home pages?
- [x] Can you load the classification page?
- [x] Can you submit a classification?
- [x] Does talk load correctly?
- [x] Can you post a talk comment?

### Review Checklist
- [x] Does it work in all major browsers: Firefox, Chrome, Edge, Safari?
- [x] Does it work on mobile?
- [x] Can you `rm -rf node_modules/ && npm install` and app works as expected?
- [x] If the component is in coffeescript, is it converted to ES6? Is it free of eslint errors? Is the conversion its own commit?
- [x] Are the tests passing locally and on Travis?

### Optional
- [x] Have you replaced any `ChangeListener` or `PromiseRenderer` components with code that updates component state?
- [x] If changes are made to the classifier, does the dev classifier still work?
- [x] Have you [resized and compressed](https://developers.google.com/web/fundamentals/performance/optimizing-content-efficiency/image-optimization) any image you've added?
- [x] Have you added in [flow type annotations](https://flowtype.org/docs/type-annotations.html)?
- [x] Have you followed the [Springer guidelines for commit messages](https://github.com/springernature/frontend-playbook/blob/master/git/git.md#commit-messages)?

jelliotartz commented:
There's also a little bit of conversation about this PR in the #pfe_improvements channel on slack.

snblickhan commented:
1. Pedant alert: you're using double quotes throughout, but then you use single quotes for #11 in the "quick guide to building a project" (probably a holdover from previous version, which is written in British English?)
2. "Navigating the Project Builder" -- the PB isn't really split into 3 sections, so this is kind of confusing. Maybe replace that first sentence with "The tabs on the left-hand side of the project builder can be split into three key areas: "Project", "Workflows" and "Subject Sets".
3. "Project" section is missing the Field Guide from its list. Similarly, a detailed explanation for Field Guide is also missing later (in the "detailed" section).
4. Project building in detail section: Checkbox, private project -- is there a private checkbox under the Project details tab? Or is it in Visibility? I don't see it when I click on the Project details tab.
5. PB in detail section: Are you planning to include information for Workflow Description, Researcher Quote, Banners & Tags?
6. About is missing an explanation for Team section.
7. Should the Visibility section have some info about applying for review?
8. What are the current default Tasks? Is it only Question and Drawing? (For example, my list is 4 things: Question, Drawing, Text & Survey). Even if Text & Survey are still Experimental features (?), I think enough projects are using these Tasks that we need to include them in documentation.
9. In the Workflow section you refer to "Drawing" tasks, but in the Task Content section you refer to "Marking" tasks. The Project Builder sorts them under Drawing, so probably best to keep that vocabulary unless the PB designation will change as well.
10. Under "Choices", the Multiple option should read "Allow Multiple" because that's what the box says in the Project Builder.
11. Under Marking (Drawing), you should change the order of items to reflect the list in the Project Builder (it's alphabetical). The Project Builder list also currently includes the bezier, column, and triangle tools -- if these are default tool options, they need to be explained here as well.
12. Circle tool isn't really an X (it's a target?) -- the X is actually what you use to cancel that mark, so this might be confusing language to use here.
13. Subject Sets "you can ~add~ upload groups of data to be classified. To do so, drag and drop items onto the drop zone in the browser and then _click the green "Upload" button._"

hrspiers commented:
Thanks for the comments @snblickhan - really useful!

@jelliotartz how shall we go about making these changes - e.g. would it be easiest if I re-edit a word doc for the page in line with the suggestions and then you update the page, or would you be happy to update directly?

jelliotartz commented:
Agreed, thanks for the comments @snblickhan! @hrspiers a word doc is fine, or I'll also mention  google docs, in case you like their format. Google docs allows multiple people to collaboratively edit a document pretty easily.

wgranger commented on app/router.cjsx:
```diff
@@ -237,7 +239,7 @@ module.exports =
     </Route>
     <Route path="lab-policies" component={require './pages/lab/help/lab-policies'} />
-    <Route path="lab-how-to" component={require './pages/lab/help'} />
+    <Route path="lab-how-to" component={HowToBuildProject} />
```
I don't think you'll ever need to hit this "/lab-how-to" route. It ends up pulling the `HowToBuildProject` without the help header, etc. The "/help" route, as defined by the `IndexRoute` component above, should be all you need here.

wgranger commented:
![screen shot 2017-12-04 at 1 14 28 pm](https://user-images.githubusercontent.com/14099077/33571269-24e1ac72-d8f5-11e7-8b25-07c07fb8365e.png)

It seems odd to navigate to the "Kitteh Zoo" example without the page being included in the help header ("How to Build a Project, Glossary, Policies, etc.). Perhaps this should be added to the header as "Example?"

jelliotartz commented:
## PR Update
- Changed the quotation marks on point 11 from single to double.
- Removed the `lab-how-to` route.
- Added a nav link to kitteh zoo, naming the tab Example.
  - Happy to adjust name, or revert, if we decide on another approach.

Holding on the rest of @snblickhan's suggestions, for either a word or google doc to work from :) 

#### [Staging branch updated](https://project-builder-documentation-rewrite.pfe-preview.zooniverse.org/).

jelliotartz commented:
[Google doc](https://docs.google.com/document/d/1lWMmtBqcZVdrChQhhfRbeXGidyaKX4g305reRh8bJak/edit) for project builder documentation text

jelliotartz commented:
## PR Update
[Staging branch](https://project-builder-documentation-rewrite.pfe-preview.zooniverse.org/help) is currently up to date with the latest changes added to the [google doc](https://docs.google.com/document/d/1lWMmtBqcZVdrChQhhfRbeXGidyaKX4g305reRh8bJak/edit?ts=5a553b6f).

jelliotartz commented:
Thanks for your suggestions, Cam. [Staging branch](https://project-builder-documentation-rewrite.pfe-preview.zooniverse.org/help) updated to address your comments.

I tested the links, and I think they're all working, but would appreciate other eyes on it to double check.

wgranger commented:
@snblickhan I'd like to get this merged soon, but there was a comment in the Google doc linked above about needing more info. in the subject set section. Any update on this?

snblickhan commented:
@wgranger Honestly, the doc still needs a bit of work. Particularly, it could use some fleshing out on the Subject Sets section, but I won't have a chance to get to that until next week because of ASM stuff. Andrea also made some good comments in the pfe_improvements Slack channel that should be added.

If someone else has a chance or wants to take a crack at it, I've made notes in the Google doc about the type of info the Subject Sets section still needs. https://docs.google.com/document/d/1lWMmtBqcZVdrChQhhfRbeXGidyaKX4g305reRh8bJak/edit

snblickhan commented:
I've updated the text & added some links to the Data section on the google doc. Need to look at ASM for a while, but if I get a chance this afternoon I'll try to look at Subject Sets.

Also, there's a reference to Aggregation documents in the Drawing section that needs a link -- not sure what this is supposed to link *to* (e.g. Github repo, etc). Suggestions welcome. Maybe @CKrawczyk has a preference for where we should point project builders?

jelliotartz commented:
[Staging branch](https://project-builder-documentation-rewrite.pfe-preview.zooniverse.org/help) updated with changes to [google doc](https://docs.google.com/document/d/1lWMmtBqcZVdrChQhhfRbeXGidyaKX4g305reRh8bJak/edit?ts=5a553b6f). Holding on:
- Subject Sets text updates
- link to aggregation documents

Please let me know if I'm forgetting/missing anything :)

CKrawczyk commented:
Drawing tool aggregation can be linked to: https://aggregation-caesar.zooniverse.org/docs

jelliotartz commented:
[Staging branch](https://project-builder-documentation-rewrite.pfe-preview.zooniverse.org/help) updated with Subject Sets text and link to aggregation docs. @snblickhan feel free to speak up if you want to add more to the Subject Sets text, otherwise I'll assume that text is good to go.

All issues in the [Google doc](https://docs.google.com/document/d/1lWMmtBqcZVdrChQhhfRbeXGidyaKX4g305reRh8bJak/edit?ts=5a6609d9) are currently resolved (I'm leaving the Subject Sets one for now), so unless anyone has something to add, this should be ready for final review.

camallen commented:
I'd be in favour of merging this as is instead of waiting for final changes, it's a mile better than what we have (which is also incomplete / wrong in parts). If we are waiting on folks to cycle back to update text, etc, instead we can revisit the new changes / updates in another PR to get this out.

eatyourgreens commented:
👍 from me.

jelliotartz commented:
Just talked with @trouille, she'd like to hold on merging this PR until @snblickhan has a chance to do some final edits of the text on Thursday.

marten commented:
@jelliotartz @snblickhan What's the status of this?

snblickhan commented:
Last I saw in Slack, @camallen had suggested merging these updates, then giving it another look a bit later on. That would work for me, as I won't have a chance to really focus on this until a hack day or something, and @trouille and I have also been talking about getting together some best practices documentation for transcription, which I could focus on at the same time.

hrspiers commented:
I'd second @camallen's suggestion - I'm keen for this to be merged, then we can make updates iteratively

wgranger commented:
Ok. I'll give it one more look (all checked out past times I've reviewed) and merge if all appears well.

hrspiers commented:
thanks Will!

wgranger commented:
Ok. Branch looks fine on multiple browsers. I'm merging this. Only comment I'll make is that it seems (I could be wrong) that the only way to access the help page is through Build a Project/How to build a project documentation. In the future, we should consider making some changes to increase visibility.
